src/events/parser.rs

```diff
@@ -0,0 +1,280 @@
+use anyhow::Result;
+use serde_json;
+use serde_yaml;
+use std::collections::BTreeMap;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ComponentType {
+    actor,
+    capability,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Annotations {
+    version: String,
+    description: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Metadata {
+    name: String,
+    annotations: Annotations,
+}
+
+fn default_link_name() -> Option<String> {
+    Some("default".to_string())
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Properties {
+    image: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    contract: Option<String>,
+    #[serde(default = "default_link_name", skip_serializing_if = "Option::is_none")]
+    link_name: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum TraitType {
+    spreadscaler,
+    linkdef,
+}
+#[derive(Debug, Serialize, Deserialize)]
+pub enum SpreadRequirements {
+    app(String),
+    zone(String),
+    ledenabled(bool),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum LinkdefValueItems {
+    uri(String),
+    address(String),
+    port(i32),
+}
+
+fn default_weight() -> Option<i32> {
```

Comment:
Nit: Rather than set the option when deserializing, it is a bit more idiomatic in Rust to create a const like `DEFAULT_WEIGHT` and then when we are accessing the data we can do `spread.weight.unwrap_or(DEFAULT_WEIGHT)`